src/VisualStudio/VisualBasic/Impl/Options/AutomationObject.vb

```diff
@@ -44,12 +44,26 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Options
             End Set
         End Property
 
+        ' This SettingStore option has now been deprecated in favor of BasicClosedFileDiagnostics.
         Public Property ClosedFileDiagnostics As Boolean
             Get
-                Return GetBooleanOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic)
+                Return ServiceFeatureOnOffOptions.IsClosedFileDiagnosticsEnabled(_optionService, LanguageNames.VisualBasic)
             End Get
             Set(value As Boolean)
-                SetBooleanOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, value)
+                ' Even though this option has been deprecated, we want to respect the setting if the user has explicitly turned off closed file diagnostics (which is the non-default value for 'ClosedFileDiagnostics').
+                ' So, we invoke the setter only for value = False.
+                If Not value Then
+                    SetIntegerOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, value:=0)
+                End If
+            End Set
+        End Property
+
+        Public Property BasicClosedFileDiagnostics As Integer
```

Comment:
Yes, if the existing bool option would have sufficed - we are now moving to a multistate integral value for this option and hence deprecating the old option. Changes to VB side doesn't change the end user experience.